csharp/src/Ice/UdpConnector.cs

```diff
@@ -9,9 +10,9 @@ internal sealed class UdpConnector : IConnector
         private readonly UdpEndpoint _endpoint;
         private readonly EndPoint _addr;
         private readonly int _hashCode;
-
-        public Connection Connect(string connectionId)
+        public Connection Connect(string connectionId, bool preferNonSecure)
         {
+            Debug.Assert(preferNonSecure == true);
```

Comment:
This assert is correct when we attempt to connect to an endpoint via a proxy, since the Ice "proxy resolution" algorithm filters out UDP endpoints when preferNonSecure = false.

Are we assuming an application is not allowed to use endpoints directly to establish a connection, or that reporting errors as Asserts for such an application is fine? If not, you could throw an ArgumentException instead of asserting.

Reply:
Yes, the idea here is that it's a bug for Ice to pass `false`. These endpoints should be filtered out during "proxy resolution".

> Are we assuming an application is not allowed to use endpoints directly to establish a connection
I don't believe we plan to allow this, no.

Reply:
Ok.